### PR TITLE
Convert Google Symptoms pipeline to pull data from BigQuery

### PR DESCRIPTION
**Pull request by nmdefries**
### Description
Switch from pulling Google Symptoms data from GitHub (now deprecated) to pulling directly from the relevant tables in BigQuery.

To reduce BigQuery usage, this pulls only the required columns (`open_covid_region_code`, `date`, and symptom columns) for dates between the export start date and the current date that do not appear in any file names in the receiving directory. An additional 6 days prior are pulled for calculating smoothed indicators.

### Changelog
Itemize code/test/documentation changes and files added/removed.
- Modifies `pull.py::pull_gs_data()` to act as a high-level function
- Creates supporting functions in `pull.py`, such as pulling a single geo type at a time, getting the list of dates to retrieve, and formatting the query string
- Adds tests for these functions
- Adds new field to `params.json` to support BigQuery API credentials

### CODE REVIEW

**Comment by jingjtang** [comment]
Is it necessary to have so many test files?

**Comment by krivard** [comment]
> Is it necessary to have so many test files?

Looks like previously this indicator had 50 test files, and now it only has 10. That's certainly an improvement, though it's still not obvious that all files are used in tests.

**Comment by jingjtang** [comment]
small linting errors

************* Module delphi_google_symptoms.run
delphi_google_symptoms/run.py:13:16: C0303: Trailing whitespace (trailing-whitespace)
delphi_google_symptoms/run.py:14:22: C0303: Trailing whitespace (trailing-whitespace)
delphi_google_symptoms/run.py:74:0: C0303: Trailing whitespace (trailing-whitespace)
delphi_google_symptoms/run.py:11:0: C0411: standard import "import time" should be placed before "import numpy as np" (wrong-import-order)
************* Module delphi_google_symptoms.pull
delphi_google_symptoms/pull.py:5:0: W0611: Unused date imported from datetime (unused-import)

**Comment by jingjtang** [comment]
@krivard Do we have the smoothing utils merged? According to #306, it seems not. Otherwise, we can directly switch to the smoothing utils.

**Comment by chinandrew** [comment]
> @krivard Do we have the smoothing utils merged? According to #306, it seems not. Otherwise, we can directly switch to the smoothing utils.

It's merged and available in delphi_utils.smooth

https://github.com/cmu-delphi/covidcast-indicators/blob/main/_delphi_utils_python/delphi_utils/smooth.py

**Comment by krivard** [comment]
> small linting errors

@chinandrew any idea why this would be failing linting when run locally, but passing in CI?

**Comment by krivard** [comment]
> @krivard Do we have the smoothing utils merged?

Not relevant to this PR -- this is just for switching to BigQuery. We'll handle switching to the smoother in a separate effort before reactivating this indicator in production.

**Comment by chinandrew** [comment]
> > small linting errors
> 
> @chinandrew any idea why this would be failing linting when run locally, but passing in CI?

That's interesting...they're failing on CI too but the check is passing. Exit code is 0 for me, which is odd. They haven't release a new version recently either, so a bit confused why thats happening.

**Comment by chinandrew** [comment]
I think I figured it out. `make lint` runs 2 lint commands, and if the first fails but the second passes, the exit code corresponds to the last command's and is 0. I'll work on a fix.

File: google_symptoms/delphi_google_symptoms/pull.py [hunk 1]
```diff
@@ -63,84 +67,299 @@ def preprocess(df, level):
...
+    df = df.rename({"date": "timestamp"}, axis=1)
 
     return df
 
-def pull_gs_data(base_url):
-    """Pull latest dataset and transform it into the appropriate format.
 
-    Pull the latest Google COVID-19 Search Trends symptoms dataset, and
-    conforms it into a dataset as described in preprocess function.
+def get_missing_dates(receiving_dir, export_start_date):
```
[the remark this answers is not in the document]
Comment (nmdefries) [reply google_symptoms/delphi_google_symptoms/pull.py:104]: Okay, I wasn't aware that's how automation works. So you're saying that I need to check the COVIDcast API directly for what dates are already available for this signal.

[the remark this answers is not in the document]
Comment (jingjtang) [reply google_symptoms/delphi_google_symptoms/pull.py:104]: That might be slow. Maybe we could have a small cache file stored in `./cache`  to record the info that you need

[the remark this answers is not in the document]
Comment (krivard) [reply google_symptoms/delphi_google_symptoms/pull.py:104]: Is this indicator using the archive differ? That automatically keeps a cache of everything that's been submitted

**Comment by nmdefries** [comment]
I fixed the other linting errors but this one isn't actually true (the import is used).

```
************* Module delphi_google_symptoms.pull
delphi_google_symptoms/pull.py:5:0: W0611: Unused date imported from datetime (unused-import)
```

**Comment by nmdefries** [comment]
Switched this to using the archive differ to manage which dates get added to the API. At the moment, the BigQuery tables process (and bill for) all rows in a table, even if filtering is used, so by default the indicator pulls all dates between "now" and the specified start date.

I've asked the Google team to partition the tables by date so that filtering actually reduces the number of rows processed. If/when that gets added, we'll want to move to pulling data from a narrower date range maybe a couple of weeks wide depending on lag and backfill.

**Comment by jingjtang** [comment]
Linting errors
```
************* Module delphi_google_symptoms.run
delphi_google_symptoms/run.py:13:16: C0303: Trailing whitespace (trailing-whitespace)
delphi_google_symptoms/run.py:14:22: C0303: Trailing whitespace (trailing-whitespace)
delphi_google_symptoms/run.py:74:0: C0303: Trailing whitespace (trailing-whitespace)
delphi_google_symptoms/run.py:11:0: C0411: standard import "import time" should be placed before "import numpy as np" (wrong-import-order)
************* Module delphi_google_symptoms.pull
delphi_google_symptoms/pull.py:5:0: W0611: Unused date imported from datetime (unused-import)
```

**Comment by jingjtang** [comment]
@nmdefries I didn't look at that previously, but how is the backfill status for google symptoms? Do we want to stick to the 5-day lag or not?

**Comment by nmdefries** [comment]
I just pulled the data and it is currently 8 days behind. Once the tables move to daily updates, we'll have a better idea of lag, but I suspect that it will be about a week.

> By the way, I compared the output files with the ones generated by the old code... They match with each other.

That's great, thanks for checking!

**Comment by nmdefries** [comment]
The tables do not appear to change over time (no backfill). I check equality of county anosmia and ageusia values on Jan 25 pulled today vs pulled 9 days ago.

**Review by jingjtang — APPROVED** [review]
lgtm！

**Pull request merged**